mmaction/datasets/pipelines/augmentations.py

```diff
@@ -42,6 +42,210 @@ def _init_lazy_if_proper(results, lazy):
+            raise ValueError('transforms must be `default` or a list of dicts'
+                             ' or iaa.Augmenter object')
+
+        if not isinstance(transforms, self.iaa.Augmenter):
+            self.aug = self.iaa.Sequential(
+                [self.imgaug_builder(t) for t in self.transforms])
+
+    def _default_transforms(self):
+        """Default transforms for imgaug."""
+
+        return [
+            dict(type='Rotate', rotate=(-30, 30)),
+            dict(
+                type='SomeOf',
+                n=(0, 3),
+                children=[
+                    dict(
+                        type='OneOf',
+                        children=[
+                            dict(type='GaussianBlur', sigma=(0, 0.5)),
+                            dict(type='AverageBlur', k=(2, 7)),
+                            dict(type='MedianBlur', k=(3, 11))
+                        ]),
+                    dict(
+                        type='OneOf',
+                        children=[
+                            dict(
+                                type='Dropout', p=(0.01, 0.1),
+                                per_channel=0.5),
+                            dict(
+                                type='CoarseDropout',
+                                p=(0.03, 0.15),
+                                size_percent=(0.02, 0.05),
+                                per_channel=0.2),
+                        ]),
+                    dict(
+                        type='AdditiveGaussianNoise',
+                        loc=0,
+                        scale=(0.0, 0.05 * 255),
+                        per_channel=0.5),
+                ]),
+        ]
+
+    def imgaug_builder(self, cfg):
+        """Import a module from imgaug.
+
+        It inherits some of :func:`build_from_cfg` logic. Use a dict object to
+        create a iaa.Augmenter object.
+
+        Args:
+            cfg (dict): Config dict. It should at least contain the key "type".
+
+        Returns:
+            obj:`iaa.Augmenter`: The constructed imgaug augmenter.
+        """
+        assert isinstance(cfg, dict) and 'type' in cfg
+        args = cfg.copy()
+
+        obj_type = args.pop('type')
+        if mmcv.is_str(obj_type):
+            obj_cls = getattr(self.iaa, obj_type)
+        elif issubclass(obj_type, self.iaa.Augmenter):
+            obj_cls = obj_type
+        else:
+            raise TypeError(
+                f'type must be a str or valid type, but got {type(obj_type)}')
+
+        if 'children' in args:
+            args['children'] = [
+                self.imgaug_builder(child) for child in args['children']
+            ]
+
+        return obj_cls(**args)
+
+    def __repr__(self):
+        repr_str = self.__class__.__name__ + f'(transforms={self.aug})'
+        return repr_str
+
+    def __call__(self, results):
+        cur_aug = self.aug.to_deterministic()
+
+        results['imgs'] = [
+            cur_aug.augment_image(frame) for frame in results['imgs']
```

Comment:
It seems that this one does not support optical flow. So an assertion would be great to prevent user accidently got wrong results for flows.

Reply:
I'm not familar with optical flow. I thought optimal flows are normal grayscale images. Why optical flow was not supported here?

Reply:
Optical flow saves offset at each pixel. Say if you flip horizontally, then the offset need to be negated other than be flipped.